crates/bevy_diagnostic/src/system_information_diagnostics_plugin.rs

```diff
@@ -69,23 +69,19 @@ pub mod internal {
         mut sysinfo: Local<Option<System>>,
     ) {
         if sysinfo.is_none() {
-            *sysinfo = Some(System::new_all());
+            *sysinfo = Some(System::new_with_specifics(
```

Comment:
`System::new_all` will retrieve a lot more information than what is being used here like all processes, network interfaces, etc... Since you only care about system memory and CPU usage, no need for everything else. Especially since you'll keep this one around! It should drop the memory impact by quite a lot.